pkg/auth/auth.go

```diff
@@ -94,15 +94,15 @@ func (c Config) GetServicePrincipalToken(podName, podNamespace, resource, aadEnd
 			return nil, err
 		}
 		defer resp.Body.Close()
-
-		if resp.StatusCode != http.StatusOK {
-			return nil, fmt.Errorf("nmi response failed with status code: %d, err: %+v", resp.StatusCode, err)
-		}
-
 		bodyBytes, err := io.ReadAll(resp.Body)
 		if err != nil {
 			return nil, err
 		}
+
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("nmi response failed with status code: %d, err: %+v", resp.StatusCode, string(bodyBytes))
```

Comment:
> err: %+v

Should this be `response body` as we are not logging an actual error? Also can we log both err and response body?

Reply:
The error is nil at this point as we're checking for `err != nil` and returning before processing the response.

Reply:
Updated the string to `response body`